### PR TITLE
Switch to stylelint-config-recess-order

### LINKED ISSUE #13: Add CSS Grid properties support
Actually, what about things that are not in that repo? I mean like background-visibility? But mainly, css grid.

### PR DESCRIPTION
**Pull request by XhmikosR**
Alternative to #50.

Closes #13, closes #50, refs #51

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -2,7 +2,8 @@
 
 module.exports = {
   'extends': [
-    'stylelint-config-standard'
+    'stylelint-config-standard',
+    'stylelint-config-recess-order'
   ],
   'plugins': [
     'stylelint-order'
```
Comment (danny007in) [inline index.js:9]: needed to remove this also `  'stylelint-order'`

File: index.js [hunk 2]
```diff
@@ -2,7 +2,8 @@
 
 module.exports = {
   'extends': [
-    'stylelint-config-standard'
+    'stylelint-config-standard',
+    'stylelint-config-recess-order'
```
Comment (danny007in) [inline index.js:6]: ` 'stylelint-config-recess-order'` not working for `.scss`

**Comment by XhmikosR** [comment]
Ping @mdo @ffoodd. Can you let me know if we should proceed with this? If yes, I'll make upstream PRs to fix the new issues later.

**Comment by ffoodd** [comment]
Seems great, the only thing that could be a concern is that we'll be depending on external source for this. @mdo's initial post is used as a reference so I guess we're pretty safe for now :)

**Comment by ffoodd** [comment]
Regarding current errors, apart from `isolation` which didn't exist in our config, the other are differences. Do we need to modify our sources, or do we make a PR to `stylelint-config-recess-oder`?

**Comment by XhmikosR** [comment]
> Seems great, the only thing that could be a concern is that we'll be depending on external source for this. @mdo's initial post is used as a reference so I guess we're pretty safe for now :)

The main idea is to stop reinventing the wheel here especially since we don't seems to do a good job maintaining the order ourselves. So, this should offload a task from us, and everyone will benefit.

> Regarding current errors, apart from `isolation` which didn't exist in our config, the other are differences. Do we need to modify our sources, or do we make a PR to `stylelint-config-recess-oder`?

Either way is fine with me as long as we move with this faster because it's the only thing currently blocking me from cutting a new major version of our config. :)

**Comment by ffoodd** [comment]
Then I guess we'd better fix our own repo and stick with the suggested config. That's fine for me 👌

**Comment by XhmikosR** [comment]
@ffoodd @mdo: this is the diff for our main branch:

```diff
 scss/_modal.scss                                             | 2 +-
 scss/_nav.scss                                               | 2 +-
 scss/mixins/_reset-text.scss                                 | 2 +-
 site/assets/scss/_layout.scss                                | 6 +++---
 site/assets/scss/_placeholder-img.scss                       | 2 +-
 site/content/docs/5.1/examples/cheatsheet/cheatsheet.css     | 8 ++++----
 site/content/docs/5.1/examples/cheatsheet/cheatsheet.rtl.css | 8 ++++----
 site/content/docs/5.1/examples/features/features.css         | 2 +-
 8 files changed, 16 insertions(+), 16 deletions(-)

diff --git a/scss/_modal.scss b/scss/_modal.scss
index 21e1258f5..803b4d811 100644
--- a/scss/_modal.scss
+++ b/scss/_modal.scss
@@ -124,8 +124,8 @@
 // Footer (for actions)
 .modal-footer {
   display: flex;
-  flex-wrap: wrap;
   flex-shrink: 0;
+  flex-wrap: wrap;
   align-items: center; // vertically center
   justify-content: flex-end; // Right align buttons with flex property because text-align doesn't work on flex items
   padding: $modal-inner-padding - $modal-footer-margin-between * .5;
diff --git a/scss/_nav.scss b/scss/_nav.scss
index ffb27d869..8660a4dbc 100644
--- a/scss/_nav.scss
+++ b/scss/_nav.scss
@@ -49,9 +49,9 @@
 
     &:hover,
     &:focus {
-      border-color: $nav-tabs-link-hover-border-color;
       // Prevents active .nav-link tab overlapping focus outline of previous/next .nav-link
       isolation: isolate;
+      border-color: $nav-tabs-link-hover-border-color;
     }
 
     &.disabled {
diff --git a/scss/mixins/_reset-text.scss b/scss/mixins/_reset-text.scss
index 354f5648b..f5bd1afec 100644
--- a/scss/mixins/_reset-text.scss
+++ b/scss/mixins/_reset-text.scss
@@ -11,7 +11,7 @@
   text-transform: none;
   letter-spacing: normal;
   word-break: normal;
-  word-spacing: normal;
   white-space: normal;
+  word-spacing: normal;
   line-break: auto;
 }
diff --git a/site/assets/scss/_layout.scss b/site/assets/scss/_layout.scss
index 03eab6f28..f83cf094c 100644
--- a/site/assets/scss/_layout.scss
+++ b/site/assets/scss/_layout.scss
@@ -4,9 +4,9 @@
 
   @include media-breakpoint-up(md) {
     display: grid;
-    gap: $grid-gutter-width;
     grid-template-areas: "sidebar main";
     grid-template-columns: 1fr 3fr;
+    gap: $grid-gutter-width;
   }
 
   @include media-breakpoint-up(lg) {
@@ -23,20 +23,20 @@
 
   @include media-breakpoint-up(md) {
     display: grid;
-    gap: inherit;
     grid-template-areas:
       "intro"
       "toc"
       "content";
     grid-template-rows: auto auto 1fr;
+    gap: inherit;
   }
 
   @include media-breakpoint-up(lg) {
     grid-template-areas:
       "intro   toc"
       "content toc";
-    grid-template-columns: 4fr 1fr;
     grid-template-rows: auto 1fr;
+    grid-template-columns: 4fr 1fr;
   }
 }
 
diff --git a/site/assets/scss/_placeholder-img.scss b/site/assets/scss/_placeholder-img.scss
index 90a29544e..6f5bbe418 100644
--- a/site/assets/scss/_placeholder-img.scss
+++ b/site/assets/scss/_placeholder-img.scss
@@ -6,8 +6,8 @@
 
 .bd-placeholder-img {
   @include font-size(1.125rem);
-  text-anchor: middle;
   user-select: none;
+  text-anchor: middle;
 }
 
 .bd-placeholder-img-lg {
diff --git a/site/content/docs/5.1/examples/cheatsheet/cheatsheet.css b/site/content/docs/5.1/examples/cheatsheet/cheatsheet.css
index 77aa0f23c..ac0342b37 100644
--- a/site/content/docs/5.1/examples/cheatsheet/cheatsheet.css
+++ b/site/content/docs/5.1/examples/cheatsheet/cheatsheet.css
@@ -99,9 +99,9 @@ body {
 @media (min-width: 1200px) {
   body {
     display: grid;
-    gap: 1rem;
-    grid-template-columns: 1fr 4fr 1fr;
     grid-template-rows: auto;
+    grid-template-columns: 1fr 4fr 1fr;
+    gap: 1rem;
   }
 
   .bd-header {
@@ -127,10 +127,10 @@ body {
   .bd-cheatsheet section,
   .bd-cheatsheet article {
     display: inherit; /* 1 */
-    gap: inherit; /* 1 */
+    grid-template-rows: auto;
     grid-template-columns: 1fr 4fr;
     grid-column: 1 / span 2;
-    grid-template-rows: auto;
+    gap: inherit; /* 1 */
   }
 
   .bd-aside {
diff --git a/site/content/docs/5.1/examples/cheatsheet/cheatsheet.rtl.css b/site/content/docs/5.1/examples/cheatsheet/cheatsheet.rtl.css
index c1a4a1ccc..a68c02559 100644
--- a/site/content/docs/5.1/examples/cheatsheet/cheatsheet.rtl.css
+++ b/site/content/docs/5.1/examples/cheatsheet/cheatsheet.rtl.css
@@ -96,9 +96,9 @@ body {
 @media (min-width: 1200px) {
   body {
     display: grid;
-    gap: 1rem;
-    grid-template-columns: 1fr 4fr 1fr;
     grid-template-rows: auto;
+    grid-template-columns: 1fr 4fr 1fr;
+    gap: 1rem;
   }
 
   .bd-header {
@@ -122,10 +122,10 @@ body {
   .bd-cheatsheet section,
   .bd-cheatsheet article {
     display: inherit; /* 1 */
-    gap: inherit; /* 1 */
+    grid-template-rows: auto;
     grid-template-columns: 1fr 4fr;
     grid-column: 1 / span 2;
-    grid-template-rows: auto;
+    gap: inherit; /* 1 */
   }
 
   .bd-aside {
diff --git a/site/content/docs/5.1/examples/features/features.css b/site/content/docs/5.1/examples/features/features.css
index 33942f7f1..cf66e7e20 100644
--- a/site/content/docs/5.1/examples/features/features.css
+++ b/site/content/docs/5.1/examples/features/features.css
@@ -30,8 +30,8 @@
 .icon-link > .bi {
   margin-top: .125rem;
   margin-left: .125rem;
-  transition: transform .25s ease-in-out;
   fill: currentColor;
+  transition: transform .25s ease-in-out;
 }
 .icon-link:hover > .bi {
   transform: translate(.25rem);

```

Should I make a PR on our repo or do you guys want to change something in stylelint-config-recess-order?

**Comment by ffoodd** [comment]
@XhmikosR Feels safe to me, this only impacts pretty recent properties (mostly Grid and SVG related) so go for it :)

**Comment by XhmikosR** [comment]
@ffoodd it's a chicken-egg problem, so perhaps we should land this, then I cut a new beta release and then we update our repos?

**Comment by ffoodd** [comment]
Agreed : :shipit:

**Pull request merged**